mongo/client_examples_test.go

```diff
@@ -0,0 +1,186 @@
+	// Given this URI, the Go driver will first communicate with localhost:27017 and use the response to discover
+	// any other nodes in the replica set.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017/?replicaSet=replset")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Call Ping to verify that the deployment is up and the Client was configured successfully.
+	if err = client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func ExampleConnect_sharded() {
+	// Create and connect a Client to a sharded deployment.
+	// The URI for a sharded cluster should specify all mongos nodes in the cluster.
+
+	// Versions 1.1.0 or later of the driver support SRV polling for mongos discovery. Prior to this version, using
+	// an SRV connection string for a sharded cluster was supported, but the hosts were only resolved during initial
+	// connection string parsing. Because individual mongos servers are not aware of each other, added/removed mongos
+	// nodes wouldn't be discovered without an application restart. SRV polling allows the driver to periodically
+	// perform DNS resolution in the background and update the its view of the cluster when nodes are added/removed.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017,localhost:27018")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_sRV() {
+	// Create and connect a Client using an SRV record.
+	// SRV records allow administrators to configure a single domain to return a list of host names.
+	// The driver will resolve SRV records prefixed with "_mongodb_tcp" and use the returned host names to
+	// build its view of the deployment.
+
+	clientOpts := options.Client().ApplyURI("mongodb+srv://server.mongodb.com")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_direct() {
+	// Create a direct connection to a host. The driver will send all requests to that host and will not
+	// automatically discover other hosts in the deployment.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017/?connect=direct")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_sCRAM() {
+	// Configure a Client with SCRAM authentication (https://docs.mongodb.com/manual/core/security-scram/).
+	// The default authentication database for SCRAM is "admin". This can be configured via the
+	// database field in the URI or the AuthSource field in the options.Credential struct.
+
+	// To configure auth via URI instead of a Credential, use
+	// "mongodb://user:password@localhost:27017/?authMechanism=scram-sha-1".
```

Comment:
Since SCRAM autodiscovery is the default, I don't think you need to show specifying authMechanism here or below.

Reply:
Removed the auth mechanism. Is there anything about SCRAM-SHA-1 vs SCRAM-SHA-256 that could cause issues if auth mechanism isn't explicitly given?